lab/lab2/js/part2-app-state.js

```diff
@@ -33,19 +33,28 @@
 ===================== */
 
 // Use the data source URL from lab 1 in this 'ajax' function:
-var downloadData = $.ajax("http://");
+var downloadData = $.ajax("https://raw.githubusercontent.com/CPLN692-MUSA611-Open-Source-GIS/datasets/master/json/world-country-capitals.json");
 
 // Write a function to prepare your data (clean it up, organize it
 // as you like, create fields, etc)
-var parseData = function() {};
+var parseData = function(data) {
+  data = JSON.parse(data);
+  return data
+};
 
 // Write a function to use your parsed data to create a bunch of
 // marker objects (don't plot them!)
-var makeMarkers = function() {};
+
+var makeMarkers = function(dat) {
+  result = [];
+  x = dat.map( a => result.push(L.marker(i[CapitalLatitude], i[CapitalLongitude])).bindPopup(i["capital"]))
+  return result };
```

Comment:
This deserves some console logging to verify behaviors - it looks to me like you're binding popups in the wrong location. Here's what might work:
```javascript
var makeMarkers = function(dat) {
  return dat.map( a => L.marker(a.CapitalLatitude, a.CapitalLongitude).bindPopup(a.capital))
};
  ```